core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2582,7 +2586,7 @@ private[spark] class CallerContext(
         val callerContext = Class.forName("org.apache.hadoop.ipc.CallerContext")
         val builder = Class.forName("org.apache.hadoop.ipc.CallerContext$Builder")
         // scalastyle:on classforname
-        val builderInst = builder.getConstructor(classOf[String]).newInstance(context)
+        val builderInst = builder.getConstructor(classOf[String]).newInstance(prepareContext(context))
```

Comment:
why?  we set the actual context when we create it and there is no way to change it.  set is just actually calling the hadoop routine.  I don't know that it matters to much right now since we always create and set but its better to do.